### PR TITLE
⚠️ CONFLICT! Lineage pull request for: skeleton

### PR DESCRIPTION
**Pull request by cisagovbot**
# Lineage Pull Request: CONFLICT #

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-python-library.git`](https://github.com/cisagov/skeleton-python-library.git)
Remote branch: `HEAD`

Check the changes in this pull request to ensure they won't cause issues with
your project.

The `lineage/skeleton` branch has **one or more unresolved merge conflicts**
that you must resolve before merging this pull request!

## How to resolve the conflicts ##

1. Take ownership of this pull request by removing any other assignees.

1. Clone the repository locally, and reapply the merge:

    ```console
    git clone git@github.com:cisagov/pe-reports.git pe-reports
    cd pe-reports
    git remote add skeleton https://github.com/cisagov/skeleton-python-library.git
    git remote set-url --push skeleton no_push
    git switch develop
    git checkout -b lineage/skeleton --track origin/develop
    git pull skeleton HEAD
    git checkout --ours -- .github/lineage.yml
    git add .github/lineage.yml
    git status
    ```

1. Review the changes displayed by the `status` command.  Fix any conflicts and
   possibly incorrect auto-merges.

1. After resolving each of the conflicts, `add` your changes to the
   branch, `commit`, and `push` your changes:

    ```console
    git add src/example/example.py tests/test_example.py 
    git commit
    git push --force --set-upstream origin lineage/skeleton
    ```

    Note that you may _append_ to the default merge commit message
    that git creates for you, but *please do not delete the existing
    content*.  It provides useful information about the merge that is
    being performed.

1. Wait for all the automated tests to pass.

1. Check the "Everything is cool" checkbox below:

    - [x] ✌️ The conflicts in this pull request have been resolved.

1. Mark this draft pull request "Ready for review".

------------

**Note:** *You are seeing this because one of this repository's maintainers has
configured [Lineage] to open pull requests.*

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

### CODE REVIEW

**Comment by schmelz21** [comment]
@mcdonnnj the re-introduction of example.py and test_example.py are conflicting and are causing check failure. Can I assume removing them first will allow for a successful merge?

**Comment by mcdonnnj** [comment]
> 
> 
> @mcdonnnj the re-introduction of example.py and test_example.py are conflicting and are causing check failure. Can I assume removing them first will allow for a successful merge?

@schmelz-ctr There are some changes to `src/example/example.py` around return codes that should be made to this project. Specifically changes related to these commits: https://github.com/cisagov/pe-reports/pull/22/commits/1df17570c764a6765e7b699bcbbc09ca42fcabc9, https://github.com/cisagov/pe-reports/pull/22/commits/ad5009c1956f476cf9968e1d28bb7da64cfb289b, and https://github.com/cisagov/pe-reports/pull/22/commits/015d47e7e9d4d4570bb42396e5a94495edb5b19d. They would be made to `src/pe_reports/report_generator.py` (and tests).

**Comment by schmelz21** [comment]
> > @mcdonnnj the re-introduction of example.py and test_example.py are conflicting and are causing check failure. Can I assume removing them first will allow for a successful merge?
> 
> @schmelz-ctr There are some changes to `src/example/example.py` around return codes that should be made to this project. Specifically changes related to these commits: [1df1757](https://github.com/cisagov/pe-reports/commit/1df17570c764a6765e7b699bcbbc09ca42fcabc9), [ad5009c](https://github.com/cisagov/pe-reports/commit/ad5009c1956f476cf9968e1d28bb7da64cfb289b), and [015d47e](https://github.com/cisagov/pe-reports/commit/015d47e7e9d4d4570bb42396e5a94495edb5b19d). They would be made to `src/pe_reports/report_generator.py` (and tests).

@mcdonnnj - ok - sounds good. looking to merge https://github.com/cisagov/pe-reports/pull/6, then address these updates to report_generator.py. - thanks

**Comment by schmelz21** [comment]
> > > @mcdonnnj the re-introduction of example.py and test_example.py are conflicting and are causing check failure. Can I assume removing them first will allow for a successful merge?
> > 
> > 
> > @schmelz-ctr There are some changes to `src/example/example.py` around return codes that should be made to this project. Specifically changes related to these commits: [1df1757](https://github.com/cisagov/pe-reports/commit/1df17570c764a6765e7b699bcbbc09ca42fcabc9), [ad5009c](https://github.com/cisagov/pe-reports/commit/ad5009c1956f476cf9968e1d28bb7da64cfb289b), and [015d47e](https://github.com/cisagov/pe-reports/commit/015d47e7e9d4d4570bb42396e5a94495edb5b19d). They would be made to `src/pe_reports/report_generator.py` (and tests).
> 
> @mcdonnnj - ok - sounds good. looking to merge #6, then address these updates to report_generator.py. - thanks

@mcdonnnj In reference to `src/example/example.py` and `tests/test_example.py`, the following updates were made for `pe-reports/src/pe_reports/` and `test/test_pe_reports.py` - a8d0a51. 

Also included into this commit was the introduction of the tests `test_division` and `test_slow_division` due to the noted updates.  `test_zero_division` was not included as it was not part of the `pe-reports/src/pe_reports/` `main()`. I set this PR to review, but these can be easily removed if they show lack of purpose.

File: src/pe_reports/report_generator.py [hunk 1]
```diff
@@ -37,6 +37,16 @@
...
+def example_div(dividend: int, divisor: int) -> float:
+    """Print some logging messages."""
+    logging.debug("This is a debug message")
+    logging.info("This is an info message")
+    logging.warning("This is a warning message")
+    logging.error("This is an error message")
+    logging.critical("This is a critical message")
+    return dividend / divisor
+
+
```
Comment (mcdonnnj) [inline src/pe_reports/report_generator.py]: The only important bits from the `src/example/example.py` file are the `main()` return changes.

```suggestion
```

File: tests/test_pe_reports.py [hunk 2]
```diff
@@ -11,6 +11,13 @@
 # cisagov Libraries
 import pe_reports.report_generator
 
+div_params = [
+    (1, 1, 1),
+    (2, 2, 1),
+    (0, 1, 0),
+    (8, 2, 4),
+]
+
```
Comment (mcdonnnj) [inline tests/test_pe_reports.py]:
```suggestion
```

File: tests/test_pe_reports.py [hunk 3]
```diff
@@ -92,5 +107,37 @@ def test_bad_log_level():
...
+    result = pe_reports.report_generator.example_div(256, 16)
+    time.sleep(4)
+    assert result == 16, "result should equal be 16"
+
+
+def test_zero_division():
+    """Verify that division by zero throws the correct exception."""
+    with pytest.raises(ZeroDivisionError):
+        pe_reports.report_generator.example_div(1, 0)
```
Comment (mcdonnnj) [inline tests/test_pe_reports.py]:
```suggestion
```

File: tests/test_pe_reports.py [hunk 4]
```diff
@@ -72,11 +79,19 @@ def test_log_levels(level):
             assert (
                 logging.root.hasHandlers() is False
             ), "root logger should not have handlers yet"
-            return_code = pe_reports.report_generator.main()
+            return_code = None
+            try:
+                pe_reports.report_generator.main()
+            except SystemExit as sys_exit:
+                return_code = sys_exit.code
+            assert return_code is None, "main() should return success"
```
Comment (mcdonnnj) [inline tests/test_pe_reports.py]: We should only need the one at the end.

```suggestion
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Nice job overall! The only things to change are removing some of the extra stuff from the skeleton. You only need to worry about mirroring the changes in how `main()` functions return and using `sys.exit()` for exiting on error.

File: src/pe_reports/report_generator.py (hunk 1 above)
Comment (schmelz21) [reply src/pe_reports/report_generator.py]: Resolved in e9b30af

File: tests/test_pe_reports.py (hunk 2 above)
Comment (schmelz21) [reply tests/test_pe_reports.py]: Resolved in e9b30af

File: tests/test_pe_reports.py (hunk 3 above)
Comment (schmelz21) [reply tests/test_pe_reports.py]: Resolved in e9b30af

File: tests/test_pe_reports.py (hunk 4 above)
Comment (schmelz21) [reply tests/test_pe_reports.py]: Resolved in e9b30af

**Comment by schmelz21** [comment]
> Nice job overall! The only things to change are removing some of the extra stuff from the skeleton. You only need to worry about mirroring the changes in how `main()` functions return and using `sys.exit()` for exiting on error.

Sounds good and will take that approach moving forward with Lineage PRs. Updates made in e9b30af.

**Review by jsf9k — APPROVED** [review]
This looks great to me.  Nice job handling this Lineage PR!

**Review by mcdonnnj — APPROVED** [review]
@schmelz-ctr Thanks for your prompt work on this. LGTM ✔

**Comment by jsf9k** [comment]
@schmelz-ctr - do you need me to merge this PR?

**Comment by schmelz21** [comment]
> @schmelz-ctr - do you need me to merge this PR?

@jsf9k - no, I was going to review with the team @ 1pm today and then do the merge if that works.

**Comment by jsf9k** [comment]
> > @schmelz-ctr - do you need me to merge this PR?
> 
> @jsf9k - no, I was going to review with the team @ 1pm today and then do the merge if that works.

That's fine.  I was just checking since it's still open; @dav3r mentioned to me that one group had a PR that we would have to merge and I couldn't remember if it was this one.

**Pull request merged**